File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Product.cs
using Feedback.Analyzer.Domain.Common.Entities;

namespace Feedback.Analyzer.Domain.Entities;

/// <summary>
/// Represents a product entity in the system
/// </summary>
public class Product : AuditableEntity
{
    /// <summary>
    /// Gets or sets the name of the product.
    /// </summary>
    public string Name { get; set; } = default!;

    /// <summary>
    /// Gets or sets the description of the product.
    /// </summary>
    public string Description { get; set; } = default!;
    
    /// <summary>
    /// Gets or sets the unique identifier of the organization associated with this product.
    /// </summary>
    public Guid OrganizationId { get; set; }

    /// <summary>
    /// Gets or sets the organization associated with this product.
    /// </summary>
    public Organization Organization { get; set; } = default!;

    /// <summary>
    /// Gets or sets the collection of customer feedback associated with this product.
    /// </summary>
    public IEnumerable<CustomerFeedback> CustomerFeedbacks { get; set; } = default!;

    public ICollection<FeedbackAnalysisWorkflow> Workflows { get; set; } = default!;
}

File: src/Insight.Box.FrontEnd/src/modules/products/models/Product.ts
/*
 * Represents a product in the organization
 */
export class Product {
    /*
     * Product id
     */
    public id!: string;

   /*
    * Product organization id
    */
    public organizationId!: string;

    /*
     * Product name
     */
    public name: string = '';

    /*
     * Product description
     */
    public description: string = '';

    constructor() {
        this.name = ' ';
        this.description = ' ';
    }
}